### PR TITLE
Add targets for net6.0 (non-breaking)

### PR DESCRIPTION
**Pull request by jon-armen**
*Description of changes:*

Adding targets for net6.0 in a non-breaking way that can be applied to the 2.x series

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jon-armen** [comment]
Without this change, we are prevented from upgrading to net6.0, as the EF Core Interceptor is broken.  Is it possible we can at least get this merged in and a release created / published to nuget.org?

**Comment by jon-armen** [comment]
@srprash , I've cleaned this up for net6.0 (nonbreaking), and got the builds passing.

**Review by srprash — APPROVED** [review]
Approving with some small comments. Thanks!

BTW, @jon-armen Did you notice any breaking behavior for supporting .Net 6 that would need to go in the 3.x release?

File: sdk/src/Core/AWSXRayRecorder.Core.csproj [hunk 1]
```diff
@@ -47,6 +47,10 @@
     <FrameworkReference Include="Microsoft.AspNetCore.App" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0'">
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
+  </ItemGroup>
+
```
[the remark this answers is not in the document]
Comment (jon-armen) [reply sdk/src/Core/AWSXRayRecorder.Core.csproj]: Updated

File: sdk/src/Handlers/SqlServer/AWSXRayRecorder.Handlers.SqlServer.csproj [hunk 2]
```diff
@@ -32,13 +32,17 @@
...
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
-    <PackageReference Include="System.Data.SqlClient">
-      <Version>4.4.0</Version>
-    </PackageReference>
+    <PackageReference Include="System.Data.SqlClient" Version="4.4.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net6.0'">
+    <PackageReference Include="System.Data.SqlClient" Version="4.4.0" />
   </ItemGroup>
```
[the remark this answers is not in the document]
Comment (jon-armen) [reply sdk/src/Handlers/SqlServer/AWSXRayRecorder.Handlers.SqlServer.csproj]: Updated

**Comment by jon-armen** [comment]
> Approving with some small comments. Thanks!
> 
> BTW, @jon-armen Did you notice any breaking behavior for supporting .Net 6 that would need to go in the 3.x release?

I didn't see anything breaking from my use cases. The biggest area I needed was around the DB Interceptor.  The contracts there are breaking between netcoreapp3.1 and net6.0

**Pull request merged**